### PR TITLE
feat: add trx-failure-report action

### PR DESCRIPTION
**Pull request by jakobalander**
Prints all failures from all trx files like:
```
###########################

Test Failures in sample.trx

###########################
Failures for UnitTest
=====================
Test description: Adding several numbers
StdOut:        Given I have entered 50 into the calculator
               done: Steps.GivenIHaveEnteredSomethingIntoTheCalculator(50) (0.0s)
               And I have entered -1 into the calculator
               done: Steps.GivenIHaveEnteredSomethingIntoTheCalculator(-1) (0.0s)
               When I press add
               done: Steps.WhenIPressAdd() (0.0s)
               Then the result should be -50 on the screen
               error: Assert.NotEqual() Failure
Message:       Test method Pickles.TestHarness.MSTest.AdditionFeature.FailToAddTwoNumbers threw exception:
               Should.Core.Exceptions.NotEqualException: Assert.NotEqual() Failure
StackTrace:    at Pickles.TestHarness.MSTest.Steps.ThenTheResultShouldBePass(Int32 result) in C:\dev\pickles-results-harness\Pickles.TestHarness\Pickles.TestHarness.MSTest\Steps.cs:line 28
               at lambda_method(Closure , IContextManager , Int32 )
               at TechTalk.SpecFlow.Bindings.MethodBinding.InvokeAction(IContextManager contextManager, Object[] arguments, ITestTracer testTracer, TimeSpan&amp; duration)
               at TechTalk.SpecFlow.Bindings.StepDefinitionBinding.Invoke(IContextManager contextManager, ITestTracer testTracer, Object[] arguments, TimeSpan&amp; duration)
               at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.ExecuteStepMatch(BindingMatch match, Object[] arguments)
               at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.ExecuteStep(StepArgs stepArgs)
               at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.OnAfterLastStep()
               at TechTalk.SpecFlow.TestRunner.CollectScenarioErrors()
               at Pickles.TestHarness.MSTest.AdditionFeature.ScenarioCleanup() in C:\dev\pickles-results-harness\Pickles.TestHarness\Pickles.TestHarness.MSTest\Addition.feature.cs:line 0
               at Pickles.TestHarness.MSTest.AdditionFeature.FailToAddTwoNumbers() in c:\dev\pickles-results-harness\Pickles.TestHarness\Pickles.TestHarness.MSTest\Addition.feature:line 18
---------------------
####################

Test Failure Summary

####################
sample.trx - UnitTest -  Adding several numbers```

### CODE REVIEW

File: trx-failure-report/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+```yaml
+on: push
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Print TRX Failures
```
Comment (sasjo) [inline trx-failure-report/README.md]: Maybe add `if: failure()` to the example?

Comment (jakobalander) [reply trx-failure-report/README.md]: good idea!

**Comment by sclausson** [comment]
@jakobalander still working on this draft?

**Comment by jakobalander** [comment]
@sclausson I've not forgotten it, I was actually doing testing on it on thursday but unfortunately the version of dotnet that we're using creates trx files of a completely different xml format, which made it unusable to us.

**Pull request closed without merging**